.github/workflows/mpi_matrix_test.yml

```diff
@@ -17,21 +20,109 @@ jobs:
+        echo ""
+        echo "Install GAMS..."
+        echo ""
+        if hash brew; then
+            wget -q https://d37drm4t2jghv5.cloudfront.net/distributions/29.1.0/macosx/osx_x64_64_sfx.exe -O gams_installer.exe
+        else
+            wget -q https://d37drm4t2jghv5.cloudfront.net/distributions/29.1.0/linux/linux_x64_64_sfx.exe -O gams_installer.exe
+        fi
+        chmod +x gams_installer.exe
+        ./gams_installer.exe -q -d gams
+        GAMS_DIR=`ls -d1 $(pwd)/gams/*/ | head -1`
+        export PATH=$PATH:$GAMS_DIR
+        export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$GAMS_DIR
+        export DYLD_LIBRARY_PATH=$DYLD_LIBRARY_PATH:$GAMS_DIR
+        cd gams/*/apifiles/Python/
+        py_ver=$(python -c 'import sys;print("%s%s" % sys.version_info[:2])')
+        gams_ver=api
+        for ver in api_*; do
+            if test ${ver:4} -le $py_ver; then
+                gams_ver=$ver
+            fi
+        done
+        cd $gams_ver
+        python setup.py -q install -noCheck
+        echo ""
+        echo "Pass key environment variables to subsequent steps"
+        echo ""
+        echo "::set-env name=PATH::$PATH"
+        echo "::set-env name=LD_LIBRARY_PATH::$LD_LIBRARY_PATH"
+        echo "::set-env name=DYLD_LIBRARY_PATH::$DYLD_LIBRARY_PATH"
+
+    - name: Install Pyomo and extensions
+      run: |
+        echo "Clone Pyomo-model-libraries..."
+        git clone --quiet https://github.com/Pyomo/pyomo-model-libraries.git
+        echo ""
+        echo "Install PyUtilib..."
+        echo ""
         pip install --quiet git+https://github.com/PyUtilib/pyutilib
-        conda install mpi4py
+        echo ""
+        echo "Install Pyomo..."
+        echo ""
         python setup.py develop
-    - name: Test with nose
+
+    - name: Set up coverage tracking
+      run: |
+        WORKSPACE=`pwd`
+        COVERAGE_PROCESS_START=${WORKSPACE}/coveragerc
+        echo "::set-env name=COVERAGE_PROCESS_START::$COVERAGE_PROCESS_START"
+        cp ${WORKSPACE}/.coveragerc ${COVERAGE_PROCESS_START}
+        echo "data_file=${WORKSPACE}/.coverage" >> ${COVERAGE_PROCESS_START}
+        SITE_PACKAGES=`python -c "from distutils.sysconfig import get_python_lib; print(get_python_lib())"`
+        if [ -z "$DISABLE_COVERAGE" ]; then
+            echo 'import coverage; coverage.process_startup()' \
+                > ${SITE_PACKAGES}/run_coverage_at_startup.pth
+        fi
+
+    - name: Download and install extensions
+      run: |
+        pyomo download-extensions
+        pyomo build-extensions
+
+    - name: Run Pyomo tests
+      run: |
+        echo "Run Pyomo tests..."
+        # Import pyomo.environ to ensure that things like the dat parser
+        # are fully set up
+        python -c 'import pyomo.environ'        
+        mpirun -np 3 nosetests -v --eval-attr="mpi and (not fragile)" \
+            pyomo `pwd`/pyomo-model-libraries
+
+    - name: Upload coverage to codecov
```

Comment:
Fascinating. I'm curious how you got this to work without the token. I'll be stealing this to see if it works on the generic Windows workflow.

Reply:
I ran across the following gem when working on something else: https://community.codecov.io/t/whitelist-github-action-servers-to-upload-without-a-token/491/24

It looks like it is a moving target: overnight they appear to have also updated the codecov GitHub Action, which might provide better support for Windows.